### PR TITLE
Remove Cython & Move to `pyproject.toml`

### PR DESCRIPTION
**Pull request by samuelcolvin**
skip change file check

Starting for on V2, first speed up build by removing cython.

### CODE REVIEW

File: README.md
Comment (samuelcolvin) [inline README.md:21]: We need to create this branch **BEFORE** this branch is merged.

File: pyproject.toml
Comment (samuelcolvin) [inline pyproject.toml:14]: @PrettyWood @hramezani I assume you're happy with this?

Let me know if not. I took your email addresses from github profiles, hope that's okay.

File: pyproject.toml
Comment (samuelcolvin) [inline pyproject.toml:17]: This stop adding the changelog to the readme on release, but I think that's fine since we're linking to the changelog with `[project.urls]`

File: pyproject.toml
Comment (PrettyWood) [reply pyproject.toml:14]: Good for me! Thank you:)

Comment (hramezani) [reply pyproject.toml:14]: Fine for me! Thank you.

Comment (samuelcolvin) [reply pyproject.toml:14]: thanks both.

File: .github/workflows/ci.yml
Comment (samuelcolvin) [inline .github/workflows/ci.yml:224]: @tiangolo I'm commenting out these tests at the moment because:
1. fastapi requires pydantc<2 so this either fails or tests pydantic v1.10 which is pointless
2. They'll start failing soon since fastapi won't support pydantic 2 when it's in development

Hope that's okay with you, if you'd like us to test FastAPI, happy to support it somehow.

**Comment by samuelcolvin** [comment]
ready I think, feedback welcome.

If nothing comes up, I'll merge once v1.10.2 is done.

File: .github/workflows/ci.yml
Comment (hramezani) [inline .github/workflows/ci.yml:222]:
```suggestion
# FastAPI has a version constraint pydantic<2.0.0, so we can't run tests, we expect them to break for now anyway
```

File: pydantic/mypy.py
Comment (hramezani) [inline pydantic/mypy.py:436]: shouldn't it be `mypy>0.930`
```suggestion
        # requires mypy>0.930
```

Comment (samuelcolvin) [reply pydantic/mypy.py:436]: no i think this is right.

File: pydantic/config.py
Comment (PrettyWood) [inline pydantic/config.py:68]: `copy_on_model_validation` should have been changed here in recent fix

File: pyproject.toml
Comment (PrettyWood) [inline pyproject.toml:99]: we could probably target 3.10 in the lint/format tools (3.11 is probably not yet supported)

File: pydantic/config.py
Comment (samuelcolvin) [reply pydantic/config.py:68]: This is wrong in one place on main, I'll fix it.

File: pyproject.toml
Comment (samuelcolvin) [reply pyproject.toml]: done, thanks.

File: README.md
Comment (samuelcolvin) [reply README.md]: done.

File: pydantic/config.py
Comment (samuelcolvin) [reply pydantic/config.py]: #4491

File: pyproject.toml
Comment (PrettyWood) [reply pyproject.toml]: maybe we can do that for mypy (and maybe others I didn't check sorry)

**Comment by samuelcolvin** [comment]
okay, I think this is ready, please review.

**Review by PrettyWood — APPROVED** [review]
LGTM!

**Pull request merged**
